core/src/main/java/com/ritense/valtimo/service/CamundaTaskService.java

```diff
@@ -117,61 +144,109 @@ public CamundaTaskService(
         this.runtimeService = runtimeService;
         this.userManagementService = userManagementService;
         this.entityManager = entityManager;
+        this.authorizationService = authorizationService;
     }
 
     @Transactional(readOnly = true)
     public CamundaTask findTaskById(String taskId) {
-        return camundaTaskRepository.findById(taskId)
+        var spec = getAuthorizationSpecification(VIEW);
+        return Optional.ofNullable(findTask(spec.and(byId(taskId))))
             .orElseThrow(() -> new TaskNotFoundException(String.format("Cannot find task %s", taskId)));
     }
 
+    @Transactional
     public void assign(String taskId, String assignee) throws IllegalStateException {
-        final CamundaTask task = findTaskById(taskId);
-        final String currentAssignee = task.getAssignee();
-        try {
-            taskService.setAssignee(task.getId(), assignee);
-            publishTaskAssignedEvent(task, currentAssignee, assignee);
-        } catch (AuthorizationException ex) {
-            throw new IllegalStateException("Cannot claim task: the user has no permission.", ex);
-        } catch (ProcessEngineException ex) {
-            throw new IllegalStateException("Cannot claim task: reason is the task doesn't exist.", ex);
+        if (assignee == null) {
+            unassign(taskId);
+        } else {
+            final CamundaTask task = runWithoutAuthorization(() -> findTaskById(taskId));
+            authorizationService.requirePermission(
+                new EntityAuthorizationRequest<>(CamundaTask.class, ASSIGNABLE, assignee, task)
```

Comment:
Not sure if this is the way to go. We're not checking permissions for the active user, we're checking permissions for another user. Maybe a separate requirePermissionForUser method, or an EntityAuthorizationForAnotherUserRequest class? Then this would keep the EntityAuthorizationRequest a little bit more clear.

Reply:
The `user` field was added to the parent `AuthorizationRequest` because we must always know for which user a authorization is being requested.

But its true that the `user` field will almost always be about the logged-in-user. That's why I created a second constructor. To keep things non-breaking and clean

Reply:
This has been refactored into a specific AuthorizationRequest type when requesting permissions for another user. See DelegateUserEntityAuthorizationRequest